jsettlers.logic/src/main/java/jsettlers/logic/player/PlayerSetting.java

```diff
@@ -1,46 +1,75 @@
 package jsettlers.logic.player;
 
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
 import java.util.Arrays;
+import java.util.Random;
 
 import jsettlers.common.CommonConstants;
 import jsettlers.common.ai.EPlayerType;
 import jsettlers.common.player.ECivilisation;
 
 /**
  * @author codingberlin
+ * @author Andreas Eberle
  */
 public class PlayerSetting {
 
-	private final EPlayerType playerType;
+	private static final short INITIAL_VERSION = 1;
+
+	private final boolean available;
+	private final Byte teamId;
```

Comment:
I used Byte here for a reason: In a fresh map, the team can be set to a specific team to specify, this cannot be changed by the user. This enables the map designer to set fixed teams (when we have the UI some day). However, the normal case is, that the team isn't set (is null), which means that the team can be chosen by the players.